src/theory/quantifiers/sygus/sygus_unif_rl.cpp

```diff
@@ -440,6 +445,18 @@ void SygusUnifRl::registerConditionalEnumerator(Node f,
     // register the conditional enumerator
     d_tds->registerEnumerator(cond, f, d_parent, true);
     d_cenum_to_stratpt[cond].clear();
+    // register lemmas to remove redundant operators from condition enumeration
+    auto it = strategy_lemmas.find(cond);
+    if (it != strategy_lemmas.end())
+    {
+      for (const Node& lemma : it->second)
+      {
+        Trace("cegis-unif-enum-debug")
+            << "* Registering lemma to remove redundand operators for " << cond
+            << " --> " << lemma << "\n";
+        d_qe->getOutputChannel().lemma(lemma);
+      }
```

Comment:
Perhaps call ```strategy_lemmas.erase(cond)``` here (we are finished processing it). This would make things not const, but I think its a good idea since conceptually we should track that we've processed it.